llvm/lib/Target/SPIRV/SPIRVEmitIntrinsics.cpp

```diff
@@ -298,8 +299,7 @@ Instruction *SPIRVEmitIntrinsics::visitLoadInst(LoadInst &I) {
 }
 
 Instruction *SPIRVEmitIntrinsics::visitStoreInst(StoreInst &I) {
-  PointerType *PTy = cast<PointerType>(I.getOperand(1)->getType());
-  if (!PTy->getPointerElementType()->isAggregateType())
+  if (!AggrStores.contains(&I))
```

Comment:
Why can't we use `I.getValueOperand()->getType()->isAggregateType()` just as the code below?

Reply:
Processing of previous instructions can change type of the operand, i.e. aggregate -> i32. As a result we get wrong behavior. It fails one LIT test transcoding/extract_insert_value.ll.